x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/transform/preview_transforms.yml

```diff
@@ -204,6 +289,35 @@ setup:
   - match: { preview.1.my_field: 42 }
   - match: { generated_dest_index.mappings.properties: {} }
 
+---
+"Test preview non-existent transform":
+  - do:
+      catch: /Transform with id \[non-existent-transform\] could not be found/
+      transform.preview_transform:
+        transform_id: "non-existent-transform"
+
+---
+"Test preview without id and body":
+  - do:
+      catch: /Either transform id param or request body is required/
+      transform.preview_transform: {}
+
+---
+"Test preview with both id and body":
+  - do:
+      catch: /transform id param \[non-existent-transform\] does not match request body id \[null\]/
+      transform.preview_transform:
```

Comment:
We should simply always disallow this. It should be an xor between the user providing the ID vs providing a config.

Reply:
+1, if I read the [docs](https://www.elastic.co/guide/en/elasticsearch/reference/current/ml-preview-datafeed.html) correctly, it sounds like a body is disallowed if id is given as part of the path.

Note: The id in the body - for the ordinary preview - is allowed for convenience. Same for dest, both id and dest are ignored in this case. I like to keep this convenience for users that switch back and forth between `PUT` and `_preview` and/or copy pasting configs into the console.